lib/TopSitesFeed.jsm

```diff
@@ -259,6 +301,37 @@ this.TopSitesFeed = class TopSitesFeed {
     }
   }
 
+  topSiteToSearchTopSite(site) {
+    if (isSearchProvider(new URL(site.url))) {
+      return {
+        searchTopSite: true,
+        label: `@${site.hostname}`,
+        ...site
+      };
+    }
+
+    return site;
+  }
+
+  /**
+   * Fetch the icon corresponding to the URL origin
+   * @param site {object} LinksCache object corresponding to a TopSite
+   */
+  async _fetchOriginTopSiteIcon(site) {
+    try {
+      const originSite = {...site, url: new URL(site.url).origin};
```

Comment:
This is probably OK, although probably doesn't need to be as complex as this (we might want to just hardcode a url in the SEARCH_PROVIDERS config with which to do a tippy-top lookup for example)